python-avd/pyavd/schema/metaschema/meta_schema_model.py

```diff
@@ -0,0 +1,559 @@
+
+    Also covers internal attributes and methods used for documentation generation.
+    All these are prefixed with underscore.
+    """
+
+    class ConvertType(str, Enum):
+        dict = "dict"
+        list = "list"
+        str = "str"
+
+    # AvdSchema field properties
+    type: Literal["list"]
+    convert_types: List[ConvertType] | None = None
+    """
+    List of types to auto-convert from.
+    For 'list of dicts' auto-conversion is supported from 'dict' if 'primary_key' is set on the list schema.
+    For other list item types conversion from dict will use the keys as list items.
+    """
+    default: List | None = None
+    """Default value"""
+    items: Annotated[AvdSchemaField, Field(discriminator="type")] | None = None
+    """Schema for list items"""
+    min_length: int | None = None
+    """Minimum list items"""
+    max_length: int | None = None
+    """Maximum list items"""
+    primary_key: str | None = Field(None, pattern=KEY_PATTERN)
+    """
+    Name of a primary key in a list of dictionaries.
+    The configured key is implicitly required and must have unique values between the list elements.
+    """
+    secondary_key: str | None = Field(None, pattern=KEY_PATTERN)
+    """Name of a secondary key, which is used with `convert_types:[dict]` in case of values not being dictionaries."""
+
+    # Type of schema docs generators to use for this schema field.
+    _table_row_generator = TableRowGenList
+    _yaml_line_generator = YamlLineGenList
+
+    @cached_property
+    def _descendant_tables(self) -> set[str]:
+        """
+        Descendant tables are used for schema docs to identify if a key should be included in a certain table.
+
+        Descendant tables returns all table names from fields below this field. Not the field itself.
+        """
+        if not self.items:
+            return set()
+
+        descendant_tables = set()
+        descendant_tables.add(self.items._table)
+        descendant_tables.update(self.items._descendant_tables)
+
+        return descendant_tables
+
+    def model_post_init(self, __context: Any) -> None:
+        """
+        Overrides BaseModel.model_post_init().
+        Runs after this model including all child models have been initilized.
+
+        Sets Internal attributes on child schema (if set):
+            - _parent_schema
+            - _is_first_list_key (except for dict)
+
+        Sets Internal attributes on grandchild schemas if child schema is a dict:
+            - _is_primary_key
+            - _is_first_list_key
+        """
+        if self.items:
+            self.items._parent_schema = self
+            if isinstance(self.items, AvdSchemaDict) and self.items.keys:
+                first_key = True
+                for key, grandchildschema in self.items.keys.items():
+                    if first_key:
+                        grandchildschema._is_first_list_key = True
+                        first_key = False
```

Comment:
Maybe need some comment in the definition of the class variable to indicate it is used in documentation to select the indentation? (I had not remembered this variable from other files :) )

Reply:
done